pac_man/pacman.py

```diff
@@ -0,0 +1,141 @@
+
+
+class Ghost(Agent):
+    def __init__(self, current_node, killable=False):
+        super().__init__(current_node)
+        self.killable = killable
+
+
+class PacMan(Agent):
+    def __init_(self, current_node):
+        super().__init__(heigt)
+
+
+class Node:
+    def __init__(self, position, children_nodes=None, reward=1):
+        self.position = position
+        self.reward = reward
+        if (children_nodes is None):
+            self.children_nodes = []
+        else:
+            self.children_nodes = children_nodes
+
+    def __str__(self):
+        return ' position : {pos} \n children nb : {children_nb} \n'.\
+            format(children_nb=str(len(self.children_nodes)),
+                   pos=self.position)
+
+
+class Board:
+    def __init__(self, board_file_name):
+        self.agents = []
+        board_nodes, board_outline = Board.construct_board(board_file_name)
+        self.board_nodes = board_nodes
+        self.board_outline = board_outline
+
+    @staticmethod
+    def construct_board(board_file_name):
+        board_outline = []
+        with open(board_file_name) as f:
+            for line in f:
+                board_outline.append([int(char) for char in line.strip()])
+        board_outline = np.array(board_outline)
+        board_row_nb, board_column_nb = board_outline.shape
+
+        board_nodes = {}
+        # Create nodes
+        for board_row_idx in range(board_row_nb):
+            for board_column_idx in range(board_column_nb):
+                if(board_outline[board_row_idx, board_column_idx] == 1):
+                    new_position = (board_row_idx, board_column_idx)
+                    new_node = Node(new_position)
+                    board_nodes[new_position] = new_node
+        # Associate nodes to children
+        for position, current_node in board_nodes.items():
+            children = []
+            row = position[0]
+            col = position[1]
+            if (row > 0 and (row - 1, col) in board_nodes.keys()):
+                children.append(board_nodes[(row - 1, col)])
+            if (row < board_row_nb and (row + 1, col) in board_nodes.keys()):
+                children.append(board_nodes[(row + 1, col)])
+            if (col > 0 and (row, col - 1) in board_nodes.keys()):
+                children.append(board_nodes[(row, col - 1)])
+            if (col < board_column_nb and
+                    (row, col + 1) in board_nodes.keys()):
+                children.append(board_nodes[(row, col + 1)])
+            current_node.children_nodes = children
+        return board_nodes, board_outline
+
+    def add_agent(self, agent):
+        target_position = agent.current_node.position
+        if (target_position in self.board_nodes):
+            target_node = self.board_nodes[agent.current_node.position]
+            self.agents.append(agent)
+        else:
+            raise OutsideOfLegalPath("""Cannot add agent
+                to invalid board position""")

```

Comment:
I am not sure but I don't think that triple quoted string are meant to be used like this for line breaking.
Either assign the string to a variable before or break the line with \